main.go

```diff
@@ -0,0 +1,451 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+    "strconv"
+	"google.golang.org/appengine"
+    "google.golang.org/appengine/datastore"
+    "google.golang.org/appengine/log"
+)
+
+type Product struct {
+    Name string
+    Barcode string
+    Description string
+    Image string
+}
+
+type Price struct {
+    Barcode string
+    Price float64
+}
+
+type Stock struct {
+    Barcode string
+    Stock int64
+}
+
+type User struct {
+    Name string
+    Balance float64
+    Cart []string
+}
+
+func main() {
+	//http.HandleFunc("/", handle)
+    http.HandleFunc("/test/", testHandle)
+    http.HandleFunc("/products", productsHandle)
+    http.HandleFunc("/products/", productsHandle)
+    http.HandleFunc("/prices", pricesHandle)
+    http.HandleFunc("/prices/", pricesHandle)
+    http.HandleFunc("/stocks", stocksHandle)
+    http.HandleFunc("/stocks/", stocksHandle)
+    http.HandleFunc("/users", usersHandle)
+    http.HandleFunc("/users/", usersHandle)
+	appengine.Main()
+}
+
+/*
+func handle(w http.ResponseWriter, r *http.Request) {
+
+    fmt.Fprintln(w, "home")
+
+}
+*/
+
+func testHandle(w http.ResponseWriter, r *http.Request) {
+    fmt.Fprintln(w, r.URL.Path)
+
+}
+
+func productsHandle(w http.ResponseWriter, r *http.Request) {
+    ctx := appengine.NewContext(r)
+    switch r.Method{
+    case "GET":
```

Comment:
please add comment on what URL will hit this case,
such as
// GET /products 

or 
// POST /products/xxxx